### ISSUE TITLE
NamedPipeClientStream.Connect(int) does not respect timeout on UNIX domain sockets

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/corefx#28791]
On Windows, if you connect to a `NamedPipeServerStream` that was created with `maxNumberOfServerInstances=1` from one process, and attempt to connect to it from another process with `NamedPipeClientStream.Connect(timeout: 0)`, it throws.

On .NET Core 2.1 on Linux, the call to `Connect` hangs indefinitely instead. Internally, it's successfully called `connect()` and is waiting on `recvmsg()`.

Repro project: 
[dotnet-pipe-server.zip](https://github.com/dotnet/corefx/files/1873538/dotnet-pipe-server.zip)

On running `repro.sh`, you should get a stack dump from the second client process, since I didn't handle any exceptions from `Connect`. But instead it just waits. Doing the equivalent manually on Windows does throw, just like full framework .NET.

**Comment by rainersigwald** [comment]
cc @stephentoub @pjanotti

**Comment by danmosemsft** [comment]
@pjanotti are you planning to look at this one (I believe there was an email thread already)?

**Comment by pjanotti** [comment]
Yes, we are already discussing this one. I will update as appropriate.

**Comment by danmosemsft** [comment]
Can you update issue @pjanotti?

**Comment by pjanotti** [comment]
There is no simple solution to this that we can take on 2.1 time frame, moving to future. The key here is that the connect from client side in domain sockets will succeed and the code will block on the receive, for now server/client needs a small protocol for the server to report that it has its desired number of clients connected.

**Comment by wfurt** [comment]
You can add Poll() with specified timeout before Receive. Based on result you either Throw() or read().
